src/langchain_google_memorystore_redis/chat_message_history.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import redis
+from typing import List, Optional
+
+from langchain_core.chat_history import BaseChatMessageHistory
+from langchain_core.messages import (
+    BaseMessage,
+    message_to_dict,
+    messages_from_dict,
+)
+
+
+class MemorystoreChatMessageHistory(BaseChatMessageHistory):
+    """Chat message history stored in a Cloud Memorystore for Redis database."""
+
+    def __init__(
+        self,
+        client: redis.Redis,
+        session_id: str,
+        ttl: Optional[int] = None,
+    ):
+        """Initializes the chat message history for Memorystore for Redis.
+
+        Args:
+            client: A redis.Redis client object.
+            session_id: A string that uniquely identifies the chat history.
+            ttl: Specifies the time in seconds after which the session will
+                expire and be eliminated from the Redis instance since the most
+                recent message is added.
+        """
+
+        self.redis = client
+        self.key = session_id
+        self.ttl = ttl
+
+        all_elements = self.redis.lrange(self.key, 0, -1)
+        self.history = messages_from_dict(
```

Comment:
The message property below should manage the loading of the messages, therefore the history property should not be used.

Reply:
The history property is just a cache of the all message history. Do you mean we can always query the Redis server for every `.messages` command? This is what I originally implemented, can you check whether commit [b033836](https://github.com/googleapis/langchain-google-memorystore-redis-python/pull/10/commits/b0338360ad1ee11017624b4674bed0db6482143a) does what you meant?

Reply:
Synced offline. Reverted the commits that adds the history property